### ISSUE TITLE
Set up CMake file to compile battle_flow_example

### ISSUE THREAD

**Issue by adinilfeld** [uchicago-cs/chiventure#918]
Currently the CMake file in `battle/examples`only compiles one example file, `battle_command_example`. I'll be modifying the CMake to compile the `battle_flow_example` file as well, which has already been written to run a sample battle using hard-coded moves. 

See also issues #839 and #892, where this problem was discovered.

**Comment by adinilfeld** [comment]
After getting advice from @firatciftci, I figured out how to update the CMake file to compile `battle_flow_example` (commit 2f5946acb901fca855708a3256b37e616a382d9d). That caused a bunch of errors + warnings, which I fixed in commit 63f9a7ef92d1fd2043491d6bec1dcb8189e9e65f. However, there's some kind of error where `battle_flow_example` ends after one turn, and with the enemy HP seeming to have overflowed (it's at -2147483578), so I'll look into how to fix that.

**Comment by adinilfeld** [comment]
After a lot of fiddling, I finally figured out what was wrong with `battle_flow_example`. Firstly, some player/enemy stats (i.e. level, defense, strength) weren't being set, so at some point during the damage calculation there was division by 0, which caused the weird overflow error. I then also had to adjust `battle_flow.c` to use a default move for the enemy -- previously the enemy wasn't given any moves, so the example battle would end in failure after the player's turn.

Now I just have to update the README to account for the fact that `battle_flow_example` is functional.
